### ISSUE TITLE
Provide alternative data last, curried api

### ISSUE THREAD

**Issue by adrian-gierakowski** [toss/es-toolkit#163]
Like in [lodash/fp](https://github.com/lodash/lodash/wiki/FP-Guide) [ramda](https://ramdajs.com/docs/#equals) (and offshoots) or [ts-belt](https://mobily.github.io/ts-belt/)

**Comment by raon0211** [comment]
Hello, we will support this feature soon.

**Comment by impurist** [comment]
Over 12 months later. Any movement of the this?

**Comment by olsonpm** [comment]
Anyone looking for a functional utility library - [Common FP](https://common-fp.org) is my solution which focuses on supporting multiple data types.  For example [mapValues](https://common-fp.org/docs/mapValues) maps over the values of an array, object, Map or Set and returns a new instance of that type.  The function names are in plain english, so newcomers aren't confused with terms like `identity function`.

It also comes with an in-browser [playground to try it out](https://common-fp.org/try-it)

**Comment by raon0211** [comment]
https://es-toolkit.dev/fp/intro.html

es-toolkit/fp supports a data-last, curried API starting from version v1.49.0.
